.github/workflows/asst-ci.yml

```diff
@@ -14,135 +17,164 @@ jobs:
       - name: Checkout Repository
         uses: actions/checkout@v2
 
-      - name: Build the ASTT Docker Image
-        run: docker build -t ghcr.io/abednigolethole/astt-cam-software .
+      - name: Check if not running against main
+        if: ${{ github.ref_name != 'main' }}
+        run: echo "Not running against main branch"
+
+      - name: Build the ASTT Docker Image for non-main branch
+        if: ${{ github.ref_name != 'main' }}
+        run: docker build -t ghcr.io/abednigolethole/astt-cam-software:${{ env.ASTT_VERSION }}-dev.c${{ github.sha }} .
+
+      - name: Build the ASTT Docker Image for main branch
+        if: ${{ github.ref_name == 'main' }}
+        run: docker build -t ghcr.io/abednigolethole/astt-cam-software:${{ env.ASTT_VERSION }} .
 
       - name: Login to GitHub Container Registry
         run: echo "${{ secrets.ASTT_PASS }}" | docker login ghcr.io --username "${{ secrets.ASTT_USR }}" --password-stdin
 
-      - name: Push the astt docker image to GitHub Container Registry
-        run: docker push ghcr.io/abednigolethole/astt-cam-software:latest
+      - name: Push the ASTT Docker Image for non-main branch
+        if: ${{ github.ref_name != 'main' }}
+        run: docker push ghcr.io/abednigolethole/astt-cam-software:${{ env.ASTT_VERSION }}-dev.c${{ github.sha }}
 
+      - name: Push the ASTT Docker Image for main branch
+        if: ${{ github.ref_name == 'main' }}
+        run: docker push ghcr.io/abednigolethole/astt-cam-software:${{ env.ASTT_VERSION }}
 
   python-lint:
     runs-on: ubuntu-20.04
     needs: build-image
 
     steps:
-    - name: Checkout code
-      uses: actions/checkout@v2
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.10'
 
-    - name: Set up Python
-      uses: actions/setup-python@v2
-      with:
-        python-version: '3.10'
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install poetry
+          poetry config virtualenvs.create false && poetry install
 
-    - name: Install dependencies
```

Comment:
I see we do not have a formal lint job, I think I have to configure it and group all the lint tools we will use. That being said, for mypy will it be a job as a stand alone?

Reply:
@Silere8  python-lint is the formal lint job. We can add mypy to the job.